cmake/flags.cmake

```diff
@@ -25,9 +25,32 @@ if (CMAKE_CXX_COMPILER_ID MATCHES GNU)
     if (ARM_TARGET EQUAL 8)
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ARM8_CXX_FLAGS}")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ARM8_CXX_FLAGS} -flax-vector-conversions")
+        add_definitions(-DHAVE_ROTR)
     elseif (ARM_TARGET EQUAL 7)
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv7-a -mfpu=neon -flax-vector-conversions")
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=armv7-a -mfpu=neon -flax-vector-conversions")
+        add_definitions(-DHAVE_ROTR)
+    elseif (XMRIG_RISCV)
+        # RISC-V-specific optimizations for maximum performance
+        # Use vector extensions (V) and bit manipulation extensions (Zba, Zbb, Zbc, Zbs)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=rv64gcv_zba_zbb_zbc_zbs")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=rv64gcv_zba_zbb_zbc_zbs")
```

Comment:
This crashes with `Illegal instruction` in my qemu. Use this as the reference: https://github.com/tevador/RandomX/blob/master/CMakeLists.txt#L185-L207

It should compile and run with `rv64gc` only, even if the performance is lower. Of course you can enable additional extensions when compiling on the real CPU (using the `try_run` function from there).